gist.go

```diff
@@ -0,0 +1,86 @@
+package model
+
+// Typed constants
+type ChannelType string
+
+// CamelCase Constant Names
+const (
+	ChannelTypeOpen    ChannelType = "O"
+	ChannelTypePrivate ChannelType = "P"
+)
+
+type Channel struct {
```

Comment:
If every field of `Channel` is not exported, how does a consumer of `model` create a new `Channel`? Does a `NewChannel` (Maybe with [Function Options](https://medium.com/@petomalina/dealing-with-optional-parameters-in-go-9780f9bfbd1d)) make sense?

Reply:
Something like `NewChannel(WithId(model.NewId()))`?

Reply:
An `id` is required and can be autogenerated, right?

I had something like `NewChannel(name, displayName string, t ChannelType, ...NewChannelOptions)` with an example usage of `NewChannel("somename", "Some Name", ChannelTypeOpen, WithHeader("Some Header"), WithPurpose("Some Purpose")` in mind.

0/5 on the naming of `WithX`